src/charm.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+
+import logging
+import subprocess
+from pathlib import Path
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus
+
+
+log = logging.getLogger(__name__)
+
+
+class UbuntuCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self.framework.observe(self.on.config_changed, self.on_config_changed)
+
+        if self.unit.is_leader():
```

Comment:
I realise this is a trivial charm, but I also think it'll end up being used as a pattern for other charms. I'm not sure running these things in `__init__` directly is a pattern we'd want to encourage. I could be wrong though, would be worth getting feedback from some others here too.

Reply:
I don't think there's anything inherently wrong with running things in `__init__` and sometimes trying to avoid it can lead to increased complexity and bugs.  I do agree that this isn't really a good example of something like that, though, so I'll change it.